framework/scene_graph/components/image.cpp

```diff
@@ -164,10 +165,16 @@ const std::vector<uint8_t> &Image::get_data() const
 	return data;
 }
 
+size_t Image::get_data_hash() const
+{
+	return data_hash;
+}
+
 void Image::clear_data()
 {
 	data.clear();
 	data.shrink_to_fit();
+	// preserve last known hash
```

Comment:
Why should the last known hash be preserved?

Reply:
In case we load the data back from GPU, not a common use case though, removed the comment to avoid confusion